### PR TITLE
feat: prettier to precommit; prettify everything

### PR DESCRIPTION
**Pull request by danisharora099**
This PR addresses https://github.com/waku-org/js-waku-examples/issues/160:
- sets up husky for pre-commit hooks
- runs `lint-staged`
   - runs `prettier` on `html,css,js,ts,jsx,tsx,json`
   - runs `eslint` on `js,ts`
- the diff includes all files in the examples prettified and eslint'd

 Note:
- eslint config used is not strict:
   - would require to address changing some `requires` -> `imports` or change rules accordingly

### CODE REVIEW

File: .husky/.gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+_
```
Comment (weboko) [inline .husky/.gitignore:1]: Do we need this file?

**Review by weboko — APPROVED** [review]

File: .husky/.gitignore (hunk 1 above)
Comment (fryorcraken) [reply .husky/.gitignore:1]: This file is not needed, husky adds `/.husky/_/,gitignore` to handle the ignoring.

**Review by fryorcraken — APPROVED** [review]

File: .husky/.gitignore (hunk 1 above)
Comment (fryorcraken) [reply .husky/.gitignore:1]: @danisharora099 not sure you saw.

Comment (danisharora099) [reply .husky/.gitignore:1]: Yes, husky adds both files to handle the ignoring. 

This was generated by husky, and the comment was resolved after a conversation with @weboko.

Comment (danisharora099) [reply .husky/.gitignore:1]: If we do remove this, an empty will be committed.

**Pull request merged**